src/statistics.jl

```diff
@@ -657,6 +721,49 @@ function Statistics.median!(
     return y
 end
 
+function Statistics.median!(
+    M::Manifold,
+    q,
+    x::AbstractVector,
+    w::AbstractVector,
+    ::WeiszfeldEstimation;
+    p0=x[1],
+    stop_iter=2000,
+    α=1.0,
+    retraction::AbstractRetractionMethod=ExponentialRetraction(),
+    inverse_retraction::AbstractInverseRetractionMethod=LogarithmicInverseRetraction(),
+    kwargs...,
+)
+    n = length(x)
+    if length(w) != n
+        throw(
+            DimensionMismatch(
+                "The number of weights ($(length(w))) does not match the number of points for the median ($(n)).",
+            ),
+        )
+    end
+    copyto!(q, p0)
+    yold = allocate_result(M, median, q)
+    ytmp = copy(yold)
+    d = zeros(n)
+    v = zero_tangent_vector(M, q)
+    wv = convert(AbstractVector, w) ./ sum(w)
+    for i in 1:stop_iter
+        d .= [distance(M, q, xi) for xi in x] # compute distances
+        # compute new weights / exclude points xi=q
+        d .= [di > 0 ? wi / di : 0 for (di, wi) in zip(d, w)]
```

Comment:
```suggestion
        d .= [di > 0 ? wi / di : zero(typeof(wi / di)) for (di, wi) in zip(d, w)]
```
Just to make it type stable.